src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -110,7 +110,7 @@ namespace Microsoft.ML.Trainers
     /// <seealso cref="Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization(MatrixFactorizationTrainer.Options)"/>
     /// <seealso cref="Options"/>
     public sealed class MatrixFactorizationTrainer : ITrainer<MatrixFactorizationModelParameters>,
-        IEstimator<MatrixFactorizationPredictionTransformer>
+        IEstimator<MatrixFactorizationPredictionTransformer>, ITrainerEstimator<MatrixFactorizationPredictionTransformer, MatrixFactorizationModelParameters>
```

Comment:
(nit) you don't need both `IEstimator` and `ITrainerEstimator`.  Since `ITrainerEstimator` inherits from `IEstimator`, you can just put `ITrainerEstimator<MatrixFactorizationPredictionTransformer, MatrixFactorizationModelParameters>` here, and remove `IEstimator<MatrixFactorizationPredictionTransformer>` for simplicity.